### PR TITLE
[script][combat-trainer] reworking the prayer mat routine to use commons

### PR DESCRIPTION
**Pull request by MahtraDR**
Preparing for a different fix for using portals as theurgy_containers, this came up. Works well now, and have tested it extensively on my cleric, as well as one other tester reporting success. Fingers crossed...

### CODE REVIEW

File: combat-trainer.lic [hunk 1]
```diff
@@ -2687,43 +2687,45 @@ class TrainerProcess
...
 
-    bput("roll #{@prayer_mat}",
-         'carefully gather up',
-         'need to be holding that first', 'not on the ground')
-    bput("put #{@prayer_mat} in my #{@prayer_mat_container}",
-         'You put', 'What were you referring to')
+    DRC.bput("kneel #{@prayer_mat}", 'You humbly kneel', 'You kneel in the water')
+    DRC.bput("kiss #{@prayer_mat}", 'You bend forward to kiss')
+
+    unless DRCI.get_item?("wine", @theurgy_supply_container)
```
Comment (rpherbig) [inline combat-trainer.lic:2715]:
```suggestion
    if DRCI.get_item?("wine", @theurgy_supply_container)
        DRC.bput("pour wine on #{@prayer_mat}", 'You quietly pour', 'Pour what', 'You set aside')
        DRCTH.empty_cleric_hands(@theurgy_supply_container)
    else
        DRC.message("*** Finishing PrayerMat early; Cannot get wine from theurgy_supply_container")
        @training_abilities.delete('PrayerMat')
    end
```
and then the shared cleanup only needs to be done once

Comment (MahtraDR) [reply combat-trainer.lic]: How's this version?

Comment (rpherbig) [reply combat-trainer.lic]: Looks good, except I don't think return on line 2721 should be there. Won't that short-circuit and not do the cleanup?

Comment (MahtraDR) [reply combat-trainer.lic]: Good catch! Thank you.

**Pull request merged**